controllers/component_build_controller_common.go

```diff
@@ -110,7 +104,8 @@ func (r *ComponentBuildReconciler) GetBuildPipelineFromComponentAnnotation(ctx c
 		return nil, err
 	}
 	if buildPipeline == nil {
-		return nil, nil
+		err = fmt.Errorf("missing or empty pipeline annotation: %s", component.Annotations[defaultBuildPipelineAnnotation])
+		return nil, boerrors.NewBuildOpError(boerrors.EMissingPipelineAnnotation, err)
```

Comment:
well I could read config map and get : default-pipeline-name
and use 'latest' for bundle
but it wouldn't be recorded in component

or I'd have to update component 

but imho, if annotation is missing that is UI/user error and we shouldn't be using something by default
but we shouldn't be hiding such thing

Reply:
so if pipeline annotation is missing
we will get default pipeline from configmap, and add pipeline annotation with the latest bundle to the component, and update component

1st reconcile will check if error was EMissingPipelineAnnotation, and return without error
2nd reconcile will already find pipeline and continue as expected

Reply:
The `pipeline` annotation is required and it can be another prerequisite of starting to handle the simple or advanced build request by Build service operator. Either fallback to docker-build pipeline or setting a default pipeline to trigger another reconcile would probably introduce side-effect. For instance, for the latter way, we have to consider whether the kubebuilder cache causes race condition.

Then, if pipeline annotation is not set, `return ctrl.Result{}, nil` and omit an Event or log.

Reply:
The `pipeline` annotation is required and it can be an additional prerequisite of starting to handle the simple or advanced build request by Build service operator. Either fallback to docker-build pipeline or setting a default pipeline to trigger another reconcile would probably introduce side-effect. For instance, for the latter way, we have to consider whether there is potential race condition. Refer to `WaitForCacheUpdate`.

Then, if pipeline annotation is not set, `return ctrl.Result{}, nil` and omit an Event or log, or both.